tini/src/seccomp_fd_notify.c

```diff
@@ -179,23 +184,60 @@ struct sock_filter net_filter[] = {
 		}
 	}
 
+	for (int i = 0;
+	     i < (unsigned short)(sizeof(check_filter) / sizeof(check_filter[0])) &&
+		 filter_idx < BPF_MAXINSNS;
+		 i++) {
+		filter[filter_idx] = check_filter[i];
+		filter_idx++;
+	}
+
 	char *is_handle_net = getenv("TITUS_SECCOMP_AGENT_HANDLE_NET_SYSCALLS");
 	char *is_handle_perf = getenv("TITUS_SECCOMP_AGENT_HANDLE_PERF_SYSCALLS");
+	char *is_traffic_steering = getenv("TITUS_SECCOMP_AGENT_HANDLE_TRAFFIC_STEERING");
 	if (is_handle_net != NULL) {
-		prog.filter = net_filter;
-		prog.len = 
-			(unsigned short)(sizeof(net_filter) / sizeof(net_filter[0]));
+		for (int i = 0;
+		     i < (unsigned short)(sizeof(net_filter) / sizeof(net_filter[0])) &&
+			 filter_idx < BPF_MAXINSNS;
+		     i++) {
+			filter[filter_idx] = net_filter[i];
+			filter_idx++;
+		}
 		PRINT_INFO("Networking system calls will be intercepted");
 	} else if (is_handle_perf != NULL) {
-		prog.filter = perf_filter;
-		prog.len = 
-			(unsigned short)(sizeof(perf_filter) / sizeof(perf_filter[0]));
+		for (int i = 0;
+		     i < (unsigned short)(sizeof(perf_filter) / sizeof(perf_filter[0])) &&
```

Comment:
1. The size of thing could be an aRRAY_SIZE macro.
2. This concatenation bit looks like it could be reused.

Reply:
Fixed.